dotCMS/src/main/java/com/liferay/portal/language/LanguageUtil.java

```diff
@@ -550,4 +551,20 @@ public static Locale getDefaultLocale(HttpServletRequest req){
 
 		return defaultLocale;
 	}
+	
+	/**
+	 * According the language and the country gets the locale, if there is no country checks the 
+	 * dotmarketing-config.properties to the default locale regarding that language. If there is no 
+	 * property set it will be set the language as the country.
+	 * 
+	 * @param languageCode
+	 * @param countryCode
+	 * @return
+	 */
+	public static String getLocale(String languageCode, String countryCode){
```

Comment:
the parameters must be final, and should not reassign them into the method code, please use a new var such as

public static String getLocale(final String languageCode, final String countryCode){
final String newCountryCode = (countryCode==null || countryCode.isEmpty())?
    Config.getStringProperty("language.default.country.mapping." + languageCode, languageCode):countryCode;

return languageCode + "_" + newCountryCode;